test/CdefTest.cc

```diff
@@ -0,0 +1,558 @@
+/*
+ * Copyright(c) 2019 Netflix, Inc.
+ * SPDX - License - Identifier: BSD - 2 - Clause - Patent
+ */
+
+/******************************************************************************
+ * @file CdefTest.cc
+ *
+ * @brief Unit test for cdef tools:
+ * * cdef_find_dir_avx2
+ * * cdef_filter_block_avx2
+ * * compute_cdef_dist_avx2
+ * * copy_rect8_8bit_to_16bit_avx2
+ * * search_one_dual_avx2
+ *
+ * @author Cidana-Wenyao
+ *
+ ******************************************************************************/
+#include <cstdlib>
+#include <string>
+#include "gtest/gtest.h"
+// workaround to eliminate the compiling warning on linux
+// The macro will conflict with definition in gtest.h
+#ifdef __USE_GNU
+#undef __USE_GNU  // defined in EbThreads.h
+#endif
+#ifdef _GNU_SOURCE
+#undef _GNU_SOURCE  // defined in EbThreads.h
+#endif
+#include "EbDefinitions.h"
+#include "aom_dsp_rtcd.h"
+#include "EbCdef.h"
+#include "util.h"
+#include "random.h"
+#include "EbUtility.h"
+
+using ::testing::make_tuple;
+using svt_av1_test_tool::SVTRandom;
+namespace {
+
+using cdef_dir_param_t =
+    ::testing::tuple<cdef_filter_block_func, cdef_filter_block_func, BlockSize,
+                     int, int>;
+
+/**
+ * @brief Unit test for cdef_filter_block_avx2
+ *
+ * Test strategy:
+ * Feed src data generated randomly and all possible input,
+ * then check the dst buffer from target function and reference
+ * function.
+ *
+ * Expect result:
+ * The dst buffer from targeted function
+ * should be identical with the values from reference function.
+ *
+ * Test coverage:
+ * Test cases:
+ * primary_strength: [0, 15] << (bd_ - 8)
+ * second_strength: 0, 1, 2, 4
+ * primary_damping: [3, 6] + (bd_ - 8)
+ * second_damping: [3, 6] + (bd_ - 8)
+ * direction: [0, 7]
+ * bitdepth: 8, 10, 12
+ *
+ */
+class CDEFBlockTest : public ::testing::TestWithParam<cdef_dir_param_t> {
+  public:
+    CDEFBlockTest() : rnd_(0, (1 << 16) - 1) {
```

Comment:
why not write to rnd_(16, false)? it is easier to read.

Reply:
The second parameter is confusing, hard to remember its signed or unsigned. I leave the range here to claim the max and min.